src/main/java/mate/academy/service/impl/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package mate.academy.service.impl;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.AuthenticationService;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userByEmail = userService.findByEmail(email);
+        if (userByEmail.isEmpty()) {
+            throw new AuthenticationException("Email or password is incorrect");
+        }
+        User user = userByEmail.get();
+        String userHashPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (user.getPassword().equals(userHashPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Email or password is incorrect");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
```

Comment:
Email should be unique. Before creating new user you should check - if this email already occupied in the db.